### ISSUE TITLE
Errors not propagated in flatMap

### ISSUE THREAD

**Issue by ajaychandran** [raquo/Airstream#42]
In the following example, errors are not propagated to the observers of `step3`.
Is this the expected behavior?

```scala
val bus  = new EventBus[Int]
val once = false
val src  = bus.events

val step1 = src.flatMap(i => EventStream.fromTry(Try(1 / i), once))
val step2 = step1.map(identity)
val step3 = step1.flatMap(EventStream.fromValue(_, once))

def logger(tag: Any): Observer[Any] = new Observer[Any] {
  def onNext(nextValue: Any): Unit     = dom.console.info("[%s] %s", tag, nextValue)
  def onError(err: Throwable): Unit    = dom.console.error("[%s] %s", tag, err)
  def onTry(nextValue: Try[Any]): Unit = nextValue.fold(onError, onNext)
}

step1.addObserver(logger(1))(unsafeWindowOwner)
step2.addObserver(logger(2))(unsafeWindowOwner)
step3.addObserver(logger(3))(unsafeWindowOwner)

bus.writer.onNext(0)
bus.writer.onNext(1)
bus.writer.onNext(0)
bus.writer.onNext(1)
```

```
[1] java.lang.ArithmeticException: / by zero
[2] java.lang.ArithmeticException: / by zero
[1] 1
[2] 1
[3] 1
[1] java.lang.ArithmeticException: / by zero
[2] java.lang.ArithmeticException: / by zero
[1] 1
[2] 1
[3] 1
````

**Comment by ajaychandran** [comment]
Could this be because `SwitchEventStream.onError` does not call `fireError?

**Comment by raquo** [comment]
Good find! Yep, this is a bug, and it seems that you're exactly right about `fireError`. Thanks for digging! I'm fixing this now.

**Comment by raquo** [comment]
Thanks again, I fixed it now in v0.11.1

**Comment by ajaychandran** [comment]
That was quick! Thanks.
